### ISSUE TITLE
Cannot use import statement outside a module (ES, CommonJS)

### ISSUE THREAD

**Issue by BLCK-B** [matrix-org/matrix-js-sdk#4347]
I tried to import matrix-js-sdk both in CommonJs and ES projects.

```
(node:22992) Warning: To load an ES module, set "type": "module" in the package.json or use the .mjs extension.
(Use `node --trace-warnings ...` to show where the warning was created)

import * as matrixcs from "./matrix";
SyntaxError: Cannot use import statement outside a module
    at wrapSafe (node:internal/modules/cjs/loader:1281:20)
    at Module._compile (node:internal/modules/cjs/loader:1321:27)
    at Module._extensions..js (node:internal/modules/cjs/loader:1416:10)
    at Module.load (node:internal/modules/cjs/loader:1208:32)
    at Module._load (node:internal/modules/cjs/loader:1024:12)
    at cjsLoader (node:internal/modules/esm/translators:348:17)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/translators:297:7)
    at ModuleJob.run (node:internal/modules/esm/module_job:222:25)
    at async ModuleLoader.import (node:internal/modules/esm/loader:316:24)
    at async asyncRunEntryPointWithESMLoader (node:internal/modules/run_main:123:5)
```

The reported fixes, type module and .mjs, do not work.

To reproduce:
https://github.com/BLCK-B/MatrixSDKissue
- cd theCode
- npm install
- npm start  *OR alternatively* node app.mjs

Node.js v20.16.0

It reportedly does not work in tsnode either (Ho Ching):
![image](https://github.com/user-attachments/assets/735e7fe7-87e4-4d34-923f-308ca42a0a9c)

Likely related:
https://github.com/matrix-org/matrix-js-sdk/issues/4284
https://github.com/matrix-org/matrix-js-sdk/issues/3145

**Comment by richvdh** [comment]
Thanks for the report.

This is actually the same thing as #4284: my "fix" in #4285 just didn't work. If you open `package.json` in https://www.npmjs.com/package/matrix-js-sdk/v/34.2.0?activeTab=code, you can see that there is no top-level `"type": "module"` setting.

I think the problem is that there are actually *two* scripts that modify `package.json` for the release process. `scripts/release/pre-release.sh` is the one that is actually used during release. `scripts/switch_package_to_release.js` is only used in CI. I have no idea why we need two of them (I don't even really know why we need one, to be honest.)

Anyway, TL;DR: let's update  `scripts/release/pre-release.sh` in the same way that #4285 did. Then let's remove  `scripts/switch_package_to_release.js` and just use  `scripts/release/pre-release.sh` in CI.

**Comment by BLCK-B** [comment]
I found that the Release Process workflow does not seem to trigger the Static Analysis, which in turn is the only trigger of `switch_package_to_release.js`, that one contains your fix.
Static Analysis:
```
on:
    pull_request: {}
    merge_group:
        types: [checks_requested]
    push:
        branches: [develop, master]
```
As for the two scripts, they seem to do the same thing. The `pre-release.sh` **is** run by Release Process.
The scripts could be merged with unified triggers. But before that, I think the solution is to add your fix `pkgJson["type"] = "module"` to `pre-release.sh`.
Why the ES module declaration is not there in the first place, I don't know.

**Comment by richvdh** [comment]
>  Why the ES module declaration is not there in the first place, I don't know.

In short, because there are a bunch of other scripts in the git repository that aren't ES modules, and adding it breaks them. But yes, fixing those scripts and adding the declaration to `package.json` in git would be a better solution.

**Comment by richvdh** [comment]
Hopefully this is fixed by #4350. We'll see after the next RC

**Comment by BLCK-B** [comment]
I see the PR caused the workflow to fail. Then richvdh tried to add it directly? Right now, it is in code on git, but not on npm.
If you manage to fix it, I'd like to know how.

**Comment by richvdh** [comment]
We're still working on this. We'll cut another RC tomorrow.

**Comment by richvdh** [comment]
sorry, forgot to update: matrix-js-sdk [v34.4.0-rc.1](https://github.com/matrix-org/matrix-js-sdk/releases/tag/v34.4.0-rc.1) was published yesterday, and this problem appears to be solved.

@BLCK-B's repro example now hits #4287, but that is a separate problem.

**Comment by lukasvaz** [comment]
im hitting  this problem  ([#4287 ](https://github.com/matrix-org/matrix-js-sdk/issues/4287))  in v34 , i  started developing just downgrading it to v32. Is there any solution available soon?.

**Comment by richvdh** [comment]
Try v34.5.0-rc.0

**Comment by vovacha** [comment]
> im hitting this problem ([#4287 ](https://github.com/matrix-org/matrix-js-sdk/issues/4287)) in v34 , i started developing just downgrading it to v32. Is there any solution available soon?.

Also downgraded to v32. It's not solved.

**Comment by richvdh** [comment]
@vovacha:

> open a new issue explaining in more detail what your problem is, and giving reproduction steps.

Please.
